lib/engine/step/tracker.rb

```diff
@@ -72,7 +72,15 @@ def lay_tile(action, extra_cost: 0, entity: nil, spender: nil)
         discount = 0
 
         entity.abilities(:tile_lay) do |ability|
-          next if !ability.hexes.include?(hex.id) || !ability.tiles.include?(tile.name)
+          next if ability.hexes.any? && (!ability.hexes.include?(hex.id) || !ability.tiles.include?(tile.name))
+
+          if ability.reachable &&
+            !@game.loading &&
+            !@game.graph.reachable_hexes(spender)[hex] &&
+            hex.name != spender.coordinates
```

Comment:
move the hex.name check first to improve performance, also can we make it inline

game_error if x &&
y &&
z

Reply:
Correction pushed.